include/libembeddedhal/frequency.hpp

```diff
@@ -279,6 +293,59 @@ class frequency
     return calculate_duty_cycle(divide(p_target), p_percent);
   }
 
+  /**
+   * @brief Calculate closest resulting frequency to target when using one of
+   * the provided dividers.
+   *
+   * @param p_target - The target frequency to achieve by using one of the
+   * dividers
+   * @param p_dividers - The set of available dividers
+   * @param p_selection_mode - The selection mode for dividers. This can
+   * restrict the dividers to result in a frequency less than or equal to the
+   * target, higher than or equal to the target or not restrict the dividers at
+   * all and get the closest value to the target.
+   * @return std::optional<frequency> - The resulting frequency or std::nullopt
+   * if no solution is found.
+   */
+  [[nodiscard]] constexpr std::optional<frequency> closest(
+    frequency p_target,
+    std::span<std::uint32_t> p_dividers,
+    selection_mode p_selection_mode)
+  {
+    auto is_applicable = [&](std::uint32_t p_candidate) -> bool {
```

Comment:
```suggestion
    auto is_applicable = [this, p_selection_mode](std::uint32_t p_candidate) -> bool {
```

Let's try to refrain from capturing everything. Unreferenced captures should be optimized away by the compiler, but we can help it along by capturing what we need.